api/server.go

```diff
@@ -0,0 +1,185 @@
+
+	writeJsonResponse(w, map[string]string{"loadTestId": u.String()})
+}
+
+func (a *API) getLoadTestById(w http.ResponseWriter, r *http.Request) (*loadtest.LoadTester, error) {
+	vars := mux.Vars(r)
+	id := vars["id"]
+	lt, ok := a.agents[id]
+	if !ok {
+		err := fmt.Errorf("Load test with id %s not found", id)
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return nil, err
+	}
+	return lt, nil
+}
+
+func (a *API) runLoadTestHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	if err = lt.Run(); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	writeJsonResponse(w, map[string]string{"message": "load test started"})
+}
+
+func (a *API) stopLoadTestHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	if err = lt.Stop(); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	writeJsonResponse(w, map[string]string{"message": "load test stopped"})
+}
+
+func (a *API) destroyLoadTestHandler(w http.ResponseWriter, r *http.Request) {
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+
+	_ = lt.Stop() // we are ignoring the error here in case the load test was previously stopped
+
+	delete(a.agents, mux.Vars(r)["id"])
+	writeJsonResponse(w, map[string]string{"message": "load test destroyed"})
+}
+
+func (a *API) getLoadTestStatusHandler(w http.ResponseWriter, r *http.Request) {
+	_, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	_ = json.NewEncoder(w).Encode(map[string]MockStatus{"status": MockStatus{}})
+}
+
+func getAmount(r *http.Request) int {
+	amountStr := r.FormValue("amount")
+	amount := 1
+
+	if amountStr != "" {
+		if a, err := strconv.ParseInt(amountStr, 10, 16); err == nil && a > 0 {
+			amount = int(a)
+		}
+	}
+	return amount
+}
+
+func (a *API) addUserHandler(w http.ResponseWriter, r *http.Request) {
+	amount := getAmount(r)
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+
+	i := 0
+	for ; i < amount; i++ {
+		if lt.AddUser() != nil {
+			break // stop on first error, result is reported as part of status
+		}
+	}
+
+	writeJsonResponse(w, map[string]interface{}{"message": fmt.Sprintf("%d users added", i+1), "status": MockStatus{}})
+}
+
+func (a *API) removeUserHandler(w http.ResponseWriter, r *http.Request) {
+	amount := getAmount(r)
+
+	lt, err := a.getLoadTestById(w, r)
+	if err != nil {
+		return
+	}
+
+	i := 0
+	for ; i < amount; i++ {
+		if err = lt.RemoveUser(); err != nil {
+			break // stop on first error, result is reported as part of status
+		}
+	}
+
+	writeJsonResponse(w, map[string]interface{}{"message": fmt.Sprintf("%d users removed", i+1), "status": MockStatus{}})
+}
+
+func SetupAPIRouter() *mux.Router {
+	router := mux.NewRouter()
+	r := router.PathPrefix("/loadtest").Subrouter()
```

Comment:
Should probably be `/loadagent` for the aforementioned reason.
Also I'd try to fix all the occurrences where we mixed up (mainly messages and API handlers).